src/libasr/codegen/asr_to_llvm.cpp

```diff
@@ -1729,7 +1729,7 @@ class ASRToLLVMVisitor : public ASR::BaseVisitor<ASRToLLVMVisitor>
     }
 
     void visit_ListCount(const ASR::ListCount_t& x) {
-        ASR::ttype_t* asr_el_type = ASRUtils::get_contained_type(ASRUtils::expr_type(x.m_arg));
+        ASR::ttype_t *asr_el_type = ASRUtils::get_contained_type(ASRUtils::type_get_past_const(ASRUtils::expr_type(x.m_arg)));
```

Comment:
I am thinking why not update the `get_contained_type()` to handle `const` such that if the type is a `const` it returns the contained type after processing the `const`. For example, I think we can update the following
https://github.com/lcompilers/lpython/blob/77baa1f9cb831a5827e560d7dcfc6ee8684868ee/src/libasr/asr_utils.h#L316-L318

to 
```cpp
        case ASR::ttypeType::Const: {
            ASR::Const_t* const_asr = ASR::down_cast<ASR::Const_t>(asr_type);
            return get_contained_type(const_asr->m_type, overload);
```

Reply:
Great idea! :+1:

Reply:
Let's do the above change.

Reply:
@Shaikh-Ubaid The above change breaks assignment:
```console
semantic error: Type mismatch in annotation-assignment, the types must be compatible
 --> ./examples/example.py:3:1
  |
3 | CONST_LIST: Const[list[i32]] = [1, 2, 3, 4, 5]
  | ^^^^^^^^^^                     ^^^^^^^^^^^^^^^ type mismatch ('i32' and 'list[i32]')


Note: Please report unclear or confusing messages as bugs at
https://github.com/lcompilers/lpython/issues.
```
```console
semantic error: Type mismatch in annotation-assignment, the types must be compatible
 --> ./examples/example.py:3:1
  |
3 | CONST_DICTIONARY: Const[dict[str, i32]] = {"a": 1, "b": 2}
  | ^^^^^^^^^^^^^^^^                          ^^^^^^^^^^^^^^^^ type mismatch ('str' and 'dict[str, i32]')


Note: Please report unclear or confusing messages as bugs at
https://github.com/lcompilers/lpython/issues.
```

Without it everything works fine.

Reply:
Ok, let's do it later then.